Add obsoletions for System.Runtime (#52014)
dotnet/runtime: docs/project/list-of-diagnostics.md

|  __`SYSLIB0021`__ | Derived cryptographic types are obsolete. Use the Create method on the base type instead. |
|  __`SYSLIB0022`__ | The Rijndael and RijndaelManaged types are obsolete. Use Aes instead. |
|  __`SYSLIB0023`__ | RNGCryptoServiceProvider is obsolete. To generate a random number, use one of the RandomNumberGenerator static methods instead. |

> |  __`SYSLIB0024`__ | Creating and unloading AppDomains is not supported and will throw PlatformNotSupportedException. |

Reviewer: I augmented the message to indicate that PNSE will be thrown.

Author: I tried to not to say that because `AppDomain.Unload` will always throw `CannotUnloadAppDomainException`, which doesn't inherit from PNSE.

Reviewer: Ah; good point. How about "...and will throw exceptions" instead?

Reviewer: ```suggestion
|  __`SYSLIB0024`__ | Creating and unloading AppDomains is not supported and throws exceptions. |
```

Reviewer 2: ```suggestion
|  __`SYSLIB0024`__ | Creating and unloading AppDomains is not supported and throws an exception. |
```

"Throws an exception" sounds better to me.
